commons/com.b2international.index.es/src/com/b2international/index/admin/EsIndexAdmin.java

```diff
@@ -0,0 +1,271 @@
+import static com.google.common.collect.Maps.newHashMapWithExpectedSize;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.regex.Pattern;
+
+import org.elasticsearch.action.admin.indices.create.CreateIndexRequestBuilder;
+import org.elasticsearch.action.admin.indices.create.CreateIndexResponse;
+import org.elasticsearch.action.admin.indices.delete.DeleteIndexResponse;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.settings.Settings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.b2international.index.Analyzed;
+import com.b2international.index.Analyzers;
+import com.b2international.index.IndexClientFactory;
+import com.b2international.index.IndexException;
+import com.b2international.index.mapping.DocumentMapping;
+import com.b2international.index.mapping.Mappings;
+import com.b2international.index.util.NumericClassUtils;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Multimap;
+import com.google.common.io.Resources;
+import com.google.common.primitives.Ints;
+import com.google.common.primitives.Primitives;
+
+/**
+ * @since 5.10
+ */
+public final class EsIndexAdmin implements IndexAdmin {
+
+	private final String name;
+	private final Mappings mappings;
+	private final Map<String, Object> settings;
+	private final Logger log;
+	private final Client client;
+
+	public EsIndexAdmin(Client client, String name, Mappings mappings, Map<String, Object> settings) {
+		this.client = client;
+		this.name = name.toLowerCase();
+		this.mappings = mappings;
+		this.settings = newHashMap(settings);
+		this.log = LoggerFactory.getLogger(String.format("index.%s", this.name));
+	}
+	
+	@Override
+	public boolean isHashSupported() {
+		return false;
+	}
+	
+	@Override
+	public Logger log() {
+		return log;
+	}
+
+	@Override
+	public boolean exists() {
+		return client().admin().indices().prepareExists(name).get().isExists();
+	}
+
+	@Override
+	public void create() {
+		if (exists()) {
+			return;
+		}
+		final CreateIndexRequestBuilder req = client().admin().indices().prepareCreate(name);
+		
+		// add mappings
+		final Multimap<String, Map<String, Object>> esMappings = HashMultimap.create();
+ 		for (DocumentMapping mapping : mappings.getMappings()) {
+			final String type = mapping.typeAsString();
+			Map<String, Object> typeMapping = ImmutableMap.of(type,
+				ImmutableMap.builder()
+					.put("_all", ImmutableMap.of("enabled", false))
+					.putAll(toProperties(mapping))
+					.build()
+			);
+			esMappings.put(type, typeMapping);
+ 		}
+ 		
+		for (String type : esMappings.keySet()) {
+			final Map<String, Object> typeMapping = esMappings.get(type).stream().sorted((m1, m2) -> -1 * Ints.compare(m1.toString().length(), m2.toString().length())).findFirst().get();
```

Comment:
Longer ones win over shorter ones. It's a compatibility hack to allow mappings to still specify both the interface and the concrete class for the index mappings.